File: examples/fibonacci.java
public class Fibonacci { 
    public static void main (String args[])  { 
		if (args.length < 1) {
			throw new RuntimeException("Must provide number for fibonacci sequence");
		}
		int n = Integer.parseInt(args[0]);
		System.out.println(fibonacci(n));
    }

	static int fibonacci(int n) {
		if (n <= 0) {
			throw new RuntimeException("Must provide a positive number for fibonacci sequence");
		} else if (n === 1) {
			return 0;
		} else if (n === 2) {
			return 1;
		} else {
			return fibonacci(n - 1) + fibonacci(n - 2);
		}
    } 
}

File: examples/fibonacci.rs
fn main() {
	let args: Vec<String> = env::args().collect();
	if args.len() < 1 {
		panic!("Must provide number for fibonacci sequence");
	}
	let n = &args[0].parse::<i32>().unwrap();
	println!("{}", fibonacci(n));
}

fn fibonacci(n: i32) -> i32 {
	if n <= 0 {
		panic!("Must provide a positive number for fibonacci sequence");
	} else if n == 1 {
		0
	} else if n == 2 {
		1
	} else {
		fibonacci(n - 1) + fibonacci(n - 2)
	}
}